### ISSUE TITLE
[Incorrect Result] torch.atan(inf) returns NaN value on dtype complex128.

### ISSUE THREAD

**Issue by dlibk** [pytorch/pytorch#122042]
### 🐛 Describe the bug

`torch.atan(inf)` returns NaN when dtype is complex128, but pi/2 on other dtypes such as float64. Please check the code below.
```python
input = torch.tensor([np.inf], dtype=torch.complex128)
out = torch.atan(input)  # tensor([nan+nanj], dtype=torch.complex128)

input = torch.tensor([np.inf], dtype=torch.float64)
out = torch.atan(input)  # tensor([1.5708], dtype=torch.float64)
```

Also, tensorflow and numpy will give pi/2 in inf input, for corresponding `arctan` functions.

### Versions

PyTorch version: 2.2.1+cu121
Is debug build: False
CUDA used to build PyTorch: 12.1
ROCM used to build PyTorch: N/A

OS: AlmaLinux 9.3 (Shamrock Pampas Cat) (x86_64)
GCC version: (GCC) 11.4.1 20230605 (Red Hat 11.4.1-2)
Clang version: 16.0.1 (https://github.com/llvm/llvm-project.git cd89023f797900e4492da58b7bed36f702120011)
CMake version: version 3.23.2
Libc version: glibc-2.34

Python version: 3.9.18 (main, Jan  4 2024, 00:00:00)  [GCC 11.4.1 20230605 (Red Hat 11.4.1-2)] (64-bit runtime)
Python platform: Linux-5.14.0-362.18.1.el9_3.x86_64-x86_64-with-glibc2.34
Is CUDA available: True
CUDA runtime version: 10.1.243
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration: 
GPU 0: NVIDIA GeForce RTX 2080 Ti
GPU 1: NVIDIA TITAN RTX
GPU 2: NVIDIA GeForce RTX 2080 Ti

Nvidia driver version: 550.54.14
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Architecture:                       x86_64
CPU op-mode(s):                     32-bit, 64-bit
Address sizes:                      46 bits physical, 48 bits virtual
Byte Order:                         Little Endian
CPU(s):                             36
On-line CPU(s) list:                0-35
Vendor ID:                          GenuineIntel
Model name:                         Intel(R) Core(TM) i9-10980XE CPU @ 3.00GHz
CPU family:                         6
Model:                              85
Thread(s) per core:                 2
Core(s) per socket:                 18
Socket(s):                          1
Stepping:                           7
CPU max MHz:                        4800.0000
CPU min MHz:                        1200.0000
BogoMIPS:                           6000.00
Flags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 cdp_l3 invpcid_single ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm mpx rdt_a avx512f avx512dq rdseed adx smap clflushopt clwb intel_pt avx512cd avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local dtherm ida arat pln pts hwp hwp_act_window hwp_epp hwp_pkg_req vnmi avx512_vnni md_clear flush_l1d arch_capabilities
Virtualization:                     VT-x
L1d cache:                          576 KiB (18 instances)
L1i cache:                          576 KiB (18 instances)
L2 cache:                           18 MiB (18 instances)
L3 cache:                           24.8 MiB (1 instance)
NUMA node(s):                       1
NUMA node0 CPU(s):                  0-35
Vulnerability Gather data sampling: Mitigation; Microcode
Vulnerability Itlb multihit:        KVM: Mitigation: VMX disabled
Vulnerability L1tf:                 Not affected
Vulnerability Mds:                  Not affected
Vulnerability Meltdown:             Not affected
Vulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable
Vulnerability Retbleed:             Mitigation; Enhanced IBRS
Vulnerability Spec rstack overflow: Not affected
Vulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl
Vulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:           Mitigation; Enhanced / Automatic IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence
Vulnerability Srbds:                Not affected
Vulnerability Tsx async abort:      Mitigation; TSX disabled

Versions of relevant libraries:
[pip3] numpy==1.26.4
[pip3] torch==2.2.1
[pip3] triton==2.2.0
[conda] Could not collect

cc @ezyang @anjali411 @dylanbespalko @mruberry @Lezcano @nikitaved @amjames

**Comment by mingfeima** [comment]
@min-jean-cho please help fix this one!

**Comment by dlibk** [comment]
It seems that this issue has been fixed on `torch 2.4.1`. Thanks for the effort!
```python
input = torch.tensor([np.inf], dtype=torch.complex128)
out = torch.atan(input)  # tensor([1.5708+0.j], dtype=torch.complex128)

input = torch.tensor([np.inf], dtype=torch.float64)
out = torch.atan(input)  # tensor([1.5708], dtype=torch.float64)
```
